app/models/reservation_scopes.rb

```diff
@@ -78,6 +78,11 @@ def self.included(base) # rubocop:disable MethodLength, AbcSize
       scope :overlaps_with_date, lambda { |date|
         where('start_date <= ? and due_date >= ?', date, date)
       }
+      scope :consecutive_with, lambda { |res|
```

Comment:
Hmmmm... I'm not sure this is the best way to figure this out. Why wouldn't we just look for all reservations with the same reserver and equipment model that are either checked out or reserved? I just feel like looking for ALL reservations that are consecutive to this one is overkill, since there might be a bunch of different equipment in a given instance and lots of users, so it seems like we're asking for more than we need. Alternatively, we can just include conditions on the reserver or equipment model in this query to make it more efficient.

More importantly, this doesn't prevent someone from creating overlapping reservations that don't exactly overlap or offset by one day. In my mind, we should look for reservations that overlap by *any* amount, find the date of first checkout and the date of last checkin for that period, and compare that against the maximum reservation length. This way just seems a bit too brittle IMHO. What do you think?